Server/Components/Pawn/Scripting/Vehicle/Natives.cpp

```diff
@@ -646,3 +646,50 @@ SCRIPT_API(GetVehicleMatrix, bool(IVehicle& vehicle, Vector3& right, Vector3& up
 	at = mat[2];
 	return true;
 }
+
+SCRIPT_API_FAILRET(GetVehicleOccupant, INVALID_PLAYER_ID, int(IVehicle& vehicle, int seat))
+{
+	IPlayer* driver = vehicle.getDriver();
+	const FlatHashSet<IPlayer*>& passengers = vehicle.getPassengers();
+	// Looking for driver
+	if (seat == 0)
+	{
+		return driver == nullptr ? INVALID_PLAYER_ID : driver->getID();
+	}
+	// Looking for a passenger
+	else
+	{
+		for (auto& passenger : passengers)
```

Comment:
> Wouldn't it make sense to store passengers as a map with seat ID as key? Lookups would be much faster then

What about some vehicles like bus which can have multiple players reporting same seat id.

Reply:
> > Wouldn't it make sense to store passengers as a map with seat ID as key? Lookups would be much faster then
> 
> What about some vehicles like bus which can have multiple players reporting same seat id.

In that case the data could be a pair with seat ID

Reply:
> > Wouldn't it make sense to store passengers as a map with seat ID as key? Lookups would be much faster then
> 
> What about some vehicles like bus which can have multiple players reporting same seat id.

Then technically it wouldn't fit all the passengers but it doesn't seem to need all of them for the current purposes (it would return the first one with that ID anyway)

Reply:
Using a map of passengers means doing lookup every tick per every received passenger sync, then decide whether to leave it, or do insertion or removal

Doing lookups and then possibly two other operation for every passenger sync VS having a pawn native and letting users decide how frequent they want it to be, I'd choose the second.

@Hual @ksenonadv

Reply:
Resolved then